tensorboard/plugins/interactive_inference/witwidget/notebook/base.py

```diff
@@ -111,6 +117,12 @@ def set_examples(self, examples):
     self.examples = [json_format.MessageToJson(ex) for ex in examples]
     self.updated_example_indices = set(range(len(examples)))
 
+  def compute_custom_distance_impl(self, index, params=None):
+    exs_for_distance = [
+        self.json_to_proto(example) for example in self.examples]
```

Comment:
should we convert to proto if the model accepts json and not tf examples?

Reply:
did we add the ability to support json for custom_predict_fn? I assumed that since WIT always uses tf.example internally keep the api of custom_predict_fn and custom_distance_fn consistent and add json support for both of them at the same time. I can also see why it is useful other way around too though (since user only deals with json for json models).

Reply:
we probably don't have that ability yet, so no need to do anything now. 

i think we should do a PR at some point soon to allow custom_predict_fns to be able to handle json (if the examples are provided as json to begin with). but you're right, its separate work.

Reply:
yes, good idea :)